### DISCUSSION TITLE
C# on mobile anyone?

### DISCUSSION THREAD

**Discussion by thirdprize** [WalletConnect/walletconnect-monorepo#1520] [Developer Support]
I have both iOS and android mobile apps written in c# and Xamarin forms.  In it, I want to open the users wallet and get the users ETH address.  This is so I can show their nfts.  How do I do this?  

I had my own deep linking solution working with metamask mobile but their deep link handling was a bit unpredictable when it came to getting back to my app.  In theory WC is a tried and trusted solution.  

I guess there is a library I need to include which calls out to the wallets and handles their responses.  How does that work? Im guessing I would need to roll my own interface as well?  If such a library doesn’t exist then I would happily write it and open source it for other c# devs.

Thanks.

**Comment by nikitaNotFound** [comment]
Hi @thirdprize! Any results with your issue?

**Reply by thirdprize** [reply]
No.  There wasn't much documentation out there and I gave up looking.  Other things to do.

**Reply by nikitaNotFound** [reply]
> No. There wasn't much documentation out there and I gave up looking. Other things to do.

BTW we use C# as well but MAUI instead of Xamarin. Maybe this NuGet will be helpful for you in case you'd like to try again.

https://www.nuget.org/packages/WalletConnect.Sign
